### ISSUE TITLE
[Analyzer] Follow Redirect

### ISSUE THREAD

**Issue by 0ssigeno** [intelowlproject/IntelOwl#1394]
## Name
Follow Redirect or something
## Type of scanner
**this can be observable, file, and docker**
Observable

## Why should we use it
There are case where the malicious observable is sent, for example, through url shortner.
We should have a way to find the **real** malicious url, so that we can analyze it in a second moment or, when we have the investigations #680, we can have a sort of preprocessing.

**Comment by tinaoberoi** [comment]
Hi @0ssigeno I am new here and would like to contribute to this issue.

**Comment by mlodic** [comment]
hey, new scanners cannot be implemented right now until the issue #1393 is solved. 
@tinaoberoi please if you want to contribute, pick one of the new analyzer issue which is a good way to start.

**Comment by spoiicy** [comment]
Hi @mlodic, regarding this issue, what I understand from the description is if there is a shortened URL passed as an observable, the analyzer should return a resolved actual malicious URL. Please correct me if I'm wrong.

**Comment by fgibertoni** [comment]
I think that the idea is to have something that checks if the submitted observable performs redirections and actively follows them. 
For each step of redirection chain (or only the final step ?) start new job with the new observable.
I think this can be achieved through Pivots.

Lmk also your thoughts @mlodic @drosetti

**Comment by mlodic** [comment]
I would rename this analyzer as "Unshorten Link" and, in practice, this would make an active request to the analyzed URL, check whether it triggers redirects (30x status codes) and get the redirect location header.
This would be a simple analyzer and needs to be run only with tlp clear because it would contact the URL directly. Plus, an option to add an HTTP proxy could be helpful for the user to avoid contacting a possible malicious URL directly.

Then, as Federico suggested, the new analyzer can be used to Pivot the extracted URL to make a new analysis with the real final URL. But that should be configured by the user because use cases may change. We could add a simple Pivot as a showcase that would trigger a classic URL reputation check after the link has been unshortened. This example would require a new playbook called "Unshorten and check reputation" or something similar.

Does it make sense to you guys?

**Comment by spoiicy** [comment]
Yes, absolutely!!!

I can definitely find the resultant URL, after multiple levels of redirections and then feed it to one of the pre-configured Pivot, like checking the URL Reputation. Along with this, we can also maybe check for Phishing sites using `CheckPhish`, as that would be quite a typical use-case for this.

**Comment by spoiicy** [comment]
@fgibertoni I think starting a new job only when we have the final URL, after reaching the end of redirection chain, would make more sense, rather than pivoting and starting a new job, after resolving to a URL on every redirection. 
Maybe @mlodic can provide some more insight here. 

Let me know guys, what is your opinion on this (˶ᵔ ᵕ ᵔ˶)

**Comment by mlodic** [comment]
Yep I agree, the analyzer could recursively handle the redirection until it doesn't get an redirection and provide the chain of redirections as a result in the analyzer output. Then we can use a Pivot to analyze the final URL

**Comment by fgibertoni** [comment]
I also agree with Matteo 😄

**Comment by spoiicy** [comment]
Post discussion with the IntelOwl team, it has been decided to integrate 2 analyzers:
1. Normal Analzyer - which basically follows redirection chain based on HTTP redirects, Meta Refresh based redirects and simple JS based redirects. 
2. Selenium based Docker Analyzer - Scenarios, when normal analyzers fail to provide reliable results, will be taken care of here. A typical browser is needed for handling some complex redirection chains, which are employed by attackers to lure users to actual malicious URLs.
